### ISSUE TITLE
feat: add feature extraction mode switch (minimal/standard/maximal/raw)

### ISSUE THREAD

**Issue by rajnavakoti** [neural-state-lab/neuro-state-monitor#32]
## Description
Add a configurable feature extraction mode that auto-selects the right compression level based on use case. Currently each classifier manually picks features — this adds a single "mode" switch for easier framework usage.

The core insight: different dataset sizes and use cases need different feature compression levels. With 200 epochs, SVM + 320 band power features works. With 50,000 epochs, raw epochs → CNN is better. Users shouldn't need to understand this tradeoff — the framework should guide them.

## Proposed Modes

| Mode | Features | Best for | N epochs needed |
|------|----------|----------|-----------------|
| `minimal` | 5 bands × 5 regions = 25 features | Real-time inference, tiny data | Any (but underfits) |
| `standard` | Band power + ERPs = ~700 features | Most use cases, 100-500 epochs | 100+ |
| `maximal` | All types (spectral + specparam + ERP + connectivity + entropy + microstates) = ~3,500 features | Research, maximum accuracy | 500+ |
| `raw` | Raw epochs (n_channels × n_times), no extraction | Deep learning (EEGNet/ShallowNet) | 1,000+ |

## Acceptance Criteria
- [ ] `shared/features/config.py` implements FeatureMode enum (minimal, standard, maximal, raw)
- [ ] `shared/features/extract.py` has a single `extract_features(epochs, mode="standard")` function that auto-selects features based on mode
- [ ] Each mode returns a consistent numpy array with appropriate feature count
- [ ] Mode can be set via YAML config (`features.mode: "standard"`)
- [ ] Auto-recommendation: if n_epochs < 100, warn user that "maximal" mode may overfit
- [ ] Documentation in config YAML explaining when to use each mode
- [ ] Existing classifier feature pipelines (encoding, addiction) updated to use mode switch
- [ ] Feature names are returned alongside the array for interpretability

## Example Usage
```python
# Simple usage
features = extract_features(epochs, mode="standard")

# Via config
config = load_config("experiments/encoding_config.yaml")
features = extract_features(epochs, mode=config["features"]["mode"])

# Auto-recommend
features, warnings = extract_features(epochs, mode="maximal")
# warnings: ["Dataset has 150 epochs but maximal mode produces 3500 features. Consider 'standard' mode."]
```

## Out of Scope
- Automatic feature selection (PCA, mutual information) — separate issue
- Hyperparameter tuning for feature extraction parameters
- Custom mode definitions (user-defined feature combinations)

## Context
Identified during learning session: different use cases (small clinical dataset vs large research pool vs real-time BCI) need different compression levels. Building blocks exist in `shared/features/` but no unified mode switch.
